src/main/java/http/server/Server.java

```diff
@@ -21,36 +24,62 @@
 public class Server {
     private Router router;
 
-    public Server(int portNum, String directory) throws IOException, InvalidRequestException, InvalidPathException, InvalidStatusCodeException {
+    public Server(int portNum, String directory) throws IOException, InvalidRequestException, InvalidPathException, InvalidStatusCodeException, NoAuthOnRouteException {
         router = configServer(new FileIO(directory));
         System.out.println("Server starting on port: " + String.valueOf(portNum) + " " + "with public directory: " + directory);
         ServerSocket socket = new ServerSocket(portNum);
         while (true) {
-        System.out.println("Waiting for client...");
-        IClient client = new Client(socket);
-        System.out.println("Client connected");
-        ClientInput clientInput = new ClientInput(client.getInputStream());
-        String rawRequest = clientInput.getRawRequestString();
-        HttpRequest httpRequest = new HttpRequest(rawRequest);
-        if (httpRequest.getContentLength() > 0) {
-            httpRequest.setBody(clientInput.getBytes(httpRequest.getContentLength()));
+            IClient client = getiClient(socket);
```

Comment:
this while loop is becoming pretty big. can you think of a good way to refactor this? from a high what is it doing?

Reply:
yeah. I'm pretty ashamed of that while loop tbh lol I'll take some time to refactor it